verde/utils.py

```diff
@@ -201,3 +202,96 @@ def maxabs(*args):
     arrays = [np.atleast_1d(i) for i in args]
     absolute = [np.abs([i.min(), i.max()]).max() for i in arrays]
     return np.max(absolute)
+
+
+def grid_to_table(grid):
+    """
+    Convert a grid to a table with the values and coordinates of each point.
+
+    Takes a 2D grid as input, extracts the coordinates and runs them through
+    :func:`numpy.meshgrid` to create a 2D table. Works for 2D grids and
+    any number of variables. Use cases includes passing gridded data to
+    functions that expect data in XYZ format such as :class:`verde.BlockReduce`
+
+    Parameters
+    ----------
+    grid : :class:`xarray.Dataset`
+        A 2D grid with a single data variable.
+
+    Returns
+    -------
+    table : :class:`pandas.DataFrame`
+        :class:`pandas.DataFrame` with coordinates and variables.
+
+    Examples
+    --------
+    >>> import xarray as xr
+    >>> import numpy as np
+    >>> temperature = xr.DataArray(np.arange(20).reshape(4,5),
+        coords = (np.arange(4), np.arange(5)), dims=['northing', 'easting'])
```

Comment:
@jessepisel you need to add `...` to get line continuations in doctests, like this:

```
>>> temperature = xr.DataArray(
...     np.arange(20).reshape(4,5),
...     coords=(np.arange(4), np.arange(5)), 
...     dims=['northing', 'easting']
... )
```